### PR TITLE
Remove gauge when type is none

### LINKED ISSUE #1416: [🐛 BUG] Metric's type="none" does not work
### What went wrong? 🤔

Even if the `type` property of the `metric` element is set to "none", the gauge shows up anyway.

### Expected Behavior

_No response_

### Steps to Reproduce Issue

The gauge should not be displayed in the following code:
```
import taipy.gui.builder as tgb
from taipy.gui import Gui

temperature = 23.2

with tgb.Page() as page:
    tgb.metric("{temperature}", type="none")

Gui(page).run()
```

### Solution Proposed

_No response_

### Screenshots

_No response_

### Runtime Environment

_No response_

### Browsers

_No response_

### OS

_No response_

### Version of Taipy

_No response_

### Additional Context

_No response_

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Resolves #1416.

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 1]
```diff
@@ -67,12 +67,19 @@ const Metric = (props: MetricProps) => {
     const theme = useTheme();
 
     const data = useMemo(() => {
+        let mode = (props.type === "none") ? "" : "gauge";
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx]: One could have used an array with join("+")

**Review by FredLL-Avaiga — DISMISSED** [review]

**Review by dinhlongviolin1 — DISMISSED** [review]
Lgtm

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 1 above)
Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx]: True. I like that!

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx]: Hum
`let mode = [(props.type === "none") ? "" : "gauge", showValue ? "number" : "", (delta !== undefined) ? "delta" : ""].join("+")`
does work... but is really hard to read.

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx]: Changed.

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 2]
```diff
@@ -67,12 +67,15 @@ const Metric = (props: MetricProps) => {
     const theme = useTheme();
 
     const data = useMemo(() => {
+        const mode = (props.type === "none") ? [] : ["gauge"];
+        showValue && mode.push("number");
+        (delta !== undefined) && mode.push("delta");
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:72]: (format document) would remove the parenthesis

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx [hunk 3]
```diff
@@ -67,12 +67,15 @@ const Metric = (props: MetricProps) => {
     const theme = useTheme();
 
     const data = useMemo(() => {
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Metric.tsx:69]: would be nice to type the data as `Data` and layout as `Partial<Layout>`
And add title to layout ;-)

**Review by FredLL-Avaiga — APPROVED** [review]

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 2 above)
Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:72]: It does not in VS...

**Review by namnguyen20999 — APPROVED** [review]
lgtm 🤩🤩🤩

File: frontend/taipy-gui/src/components/Taipy/Metric.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:72]: it does in my VS ...

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Metric.tsx:72]: Show me the settings.json one of these days.
Honestly, I prefer to keep the parenthesis that improve the reading at no runtime cost.

**Pull request merged**